docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,267 @@
+
+In order to use Gatsby, you have to install it:
+
+```shell
+npm install --save gatsby
+```
+
+> **Note**: rather than using the `gatsby new` command like you would initializing a new Gatsby site, this will install Gatsby as a dependency in your project
+
+After installation, the key things that need to change are:
+
+1. protecting any calls to browser based APIs (if there are any)
+
+1. converting routes into pages in the `/pages` directory
+
+The following sections explain the above steps as well as other changes that you might need to make depending on the complexity of your app. A default Create React App project is able to run with just the above steps ([see an example](https://twitter.com/gill_kyle/status/1136666618224730112)).
+
+### Project Structure
+
+To show some of the differences of how your project structure could differ by moving to Gatsby, a default Create React App project looks something like this:
+
+```diff
+├── .git
+├── .gitignore
+├── README.md
+├── node_modules
+├── package.json
+├── src
+│   ├── App.css
+│   ├── App.js
+│   ├── App.test.js
+│   ├── index.css
+│   ├── index.js
+│   ├── logo.svg
+│   └── serviceWorker.js
+└── yarn.lock
+```
+
+Whereas a default Gatsby project will look something like this, files that are different between Create React App and Gatsby are highlighted):
+
+```diff
+├── .git
+├── .gitignore
+// highlight-start
+├── .prettierrc
+├── LICENSE
+// highlight-end
+├── README.md
+// highlight-start
+├── gatsby-browser.js
+├── gatsby-config.js
+├── gatsby-node.js
+├── gatsby-ssr.js
+// highlight-end
+├── node_modules
+├── package.json
+// highlight-start
+├── src
+│   ├── components
+│   ├── images
+│   └── pages
+// highlight-end
+└── yarn.lock
+```
+
+The structure of a [Gatsby project](/docs/gatsby-project-structure/) adds some additional configuration files to hook into specific Gatsby APIs for the [browser](/docs/browser-apis/) and for [server-side rendering](/docs/ssr-apis/), though much of the project structure is similar enough to feel comfortable quickly.
+
+An important difference to note is the `/pages` folder in the Gatsby project structure where components will automatically be turned into static pages by Gatsby. This is discussed more in the [routing](#routing) section below.
+
+The `src/pages/index.js` file in Gatsby is a little different from the `src/index.js` file in Create React App where the root React element mounts, because it is a sibling to other pages in the app -- rather than a parent like Create React App. This means in order to share state or wrap multiple pages you should use the [wrapRootElement]() Gatsby API.
+
+> **Note**: You could import your top level `App.js` in a file at `/src/pages/index.js` and your project will likely run with the `gatsby develop` command already, though you would lose many of the benefits like code splitting and server-side rendering that Gatsby uses for pages
```

Comment:
> though you would lose many of the benefits like code splitting and server-side rendering that Gatsby uses for pages

This is presuming that routing or something is setup, right? Otherwise this isn't _necessarily_ true--but I see your point!

Reply:
Yeah maybe this is diving into a little too much minutia.. might just remove it since it doesn't really add a ton and might just confuse people 🤔  I didn't love this note since it's a little unclear, but the idea was that yeah. If your app runs in CRA it'd run just the same by importing it in the `index.js` in the pages folder.

Reply:
Simpler is probably better, and you could always add something back in if we get questions about it.